src/Mapper/ArrayToObjectMapper.php

```diff
@@ -94,10 +101,11 @@ private function getCaster(ReflectionProperty $property): ?Caster
         }
 
         if (! $castWith) {
-            return match($property->getType()->getName()) {
+            return match($type = $property->getType()->getName()) {
                 'int' => new IntegerCaster(),
                 'float' => new FloatCaster(),
                 'bool' => new BooleanCaster(),
+                DateTimeImmutable::class, DateTime::class, DateTimeInterface::class => $this->createDateCaster($type, $property),
```

Comment:
I don't think a hardcoded mapping will get us very far in the long run. But I'm also unsure about how to handle it yet.

I think I want all mappers (more than Date mappers) to be discovered automatically, and determine their mapping types on their return type.

I'm writing it here for some background info, it's totally fine to have it like this in the PR, we can refactor later.

Reply:
Completely agree - wanted to keep the scope small for now